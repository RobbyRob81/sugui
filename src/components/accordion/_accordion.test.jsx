import * as React from 'react';
import { customRender, fireEvent } from '../../../test/test-utils';
import Accordion from '.';

it('should render without throwing an error', () => {
  const { container } = customRender(<Accordion>Test</Accordion>);
  expect(container.firstChild).toMatchSnapshot();
});

it('should hide content by default', () => {
  const { getByText } = customRender(<Accordion>Test</Accordion>);
  expect(getByText(/Test/i)).not.toBeVisible();
});

it('should display content if opened by default is set', () => {
  const { getByText } = customRender(<Accordion openByDefault>Test</Accordion>);
  expect(getByText(/Test/i)).toBeVisible();
});

it('calls "onChange" prop when the opened state changes', () => {
  const onChange = jest.fn();
  const { getByText } = customRender(<Accordion title="title" onChange={onChange}>Test</Accordion>);

  fireEvent.click(getByText(/title/i));
  expect(onChange).toHaveBeenCalled();
});
