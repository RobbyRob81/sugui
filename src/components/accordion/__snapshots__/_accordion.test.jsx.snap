// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<section
  class="sc-bwzfXH bbJFoS sc-bdVaJa dWiwem"
>
  <div
    class="sc-htpNat jrSLbR"
  >
    
    <svg
      class="sc-ifAKCX dnBYYy"
      height="31"
      viewBox="0 0 32 31"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M0 0h32v30.974H0z"
          fill="none"
          stroke="none"
        />
        <path
          d="M27.333 10.97L16 21.94 4.667 10.97"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </g>
    </svg>
  </div>
  <div
    class="sc-bxivhb kQaonn"
    style="height: 0px;"
  >
    Test
  </div>
</section>
`;
